File: solutions/csharp/resistor-color-trio/1/ResistorColorTrio.cs
using System;

public static class ResistorColorTrio
{
    private static ulong Value(string color) => color.ToLower() switch
        {
            "black" => 0,
            "brown" => 1,
            "red" => 2,
            "orange" => 3,
            "yellow" => 4,
            "green" => 5,
            "blue" => 6,
            "violet" => 7,
            "grey" => 8,
            "white" => 9,
            _ => throw new ArgumentException("unknown color")
        };

    private static ulong Value(string[] colors)
    {
        var values = colors.Select(Value).ToArray();
        var significantValue = values[0] * 10UL + values[1];
        return significantValue * (ulong)Math.Pow(10, values[2]);
    }

    private static (ulong value, string prefix) MetricReduce(ulong initialValue) =>
        initialValue switch
        {
                >= 1000000000UL => (initialValue / 1000000000UL, "giga"),
                >= 1000000 => (initialValue / 1000000, "mega"),
                >= 1000 => (initialValue / 1000, "kilo"),
                _ => (initialValue, string.Empty)
        };
    
    public static string Label(string[] colors)
    {
        var (reducedValue, prefix) = MetricReduce(Value(colors));
        return $"{reducedValue} {prefix}ohms";
    }
    
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
import math


def color_value(color: str) -> int:
    return [
        "black", "brown", "red", "orange", "yellow",
        "green", "blue", "violet", "grey", "white"
    ].index(color)

def scale_value(value: int, unit: str) -> (int, str):
    for mult, prefix in [
        (1_000_000_000, "giga"),
        (1_000_000, "mega"),
        (1_000, "kilo"),
    ]:
        if value > mult:
            return (value / mult, f"{prefix}{unit}")
    return (value, unit)
    

def label(colors):
    match colors:
        case [first, second, mult, *_]:
            base_value = 10 * color_value(first) + color_value(second)
            mult = math.pow(10, color_value(mult))
            value = base_value * mult
            scaled_value, units = scale_value(value, "ohms")
            return f"{scaled_value:.0f} {units}"
        case _:
            raise ValueError("not a trio")